### DISCUSSION TITLE
Do you have plans to add the 'since_time' parameter to LogParams?

### DISCUSSION THREAD

**Discussion by sarub0b0** [kube-rs/kube#1339] [General]
I would like to implement a [stern](https://github.com/stern/stern.git) like feature in [kubetui](https://github.com/sarub0b0/kubetui.git) that enables the resumption of logging in case of errors. 
For this purpose, I intend to use the 'since_time' parameter.

**Comment by clux** [comment]
We just model the upstream API for kube-core param stuff. `LogParams::since_seconds` is probably the closest thing to that.

**Reply by sarub0b0** [reply]
Thenk you for the explanation. 
I believe Kubernetes supports the sinceTime query parameter. Could you provide a bit more explanation about why LogParams does not have since_time?

**Reply by Dav1dde** [reply]
But it doesn't in the API: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/

ctrl+f for `Read Log`

**Reply by sarub0b0** [reply]
Thank you for the documentation link. Indeed, I can see that `since_time` is not present. 
I appreciate the clarification.

**Reply by nightkr** [reply]
From the `sinceSeconds` docs:

> Only one of sinceSeconds or sinceTime may be specified.

This kind of implies that a `sinceTime` at least existed at some point...

**Reply by nightkr** [reply]
Seems to still be there in the current master: https://github.com/kubernetes/kubernetes/blob/5ce0bd95cc6e6cf0333f6e58623344d9562ba7e2/pkg/apis/core/types.go#L5177-L5181

Not exactly the first time K8s had completely undocumented flags or options...

**Reply by clux** [reply]
It is supported in kubectl also. Found this [rotten openapi bug about it](https://github.com/kubernetes/kubernetes/issues/108498) 🙃 

Did a quick PR to try to do this in https://github.com/kube-rs/kube/pull/1342/files . Might need some review and ideas, but that at least works.

**Reply by sarub0b0** [reply]
Thank you so much for this support.
